internal/pagination/books.go

```diff
@@ -0,0 +1,172 @@
+package pagination
+
+import (
+	"bookshop/internal/models"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+const (
+	formatParam   = "format"
+	categoryParam = "category"
+	minPriceParam = "min_price"
+	maxPriceParam = "max_price"
+)
+
+type BookPaginationData struct {
```

Comment:
Can we separate filters from common pagination struct? It can be just something like BookFilter.